### PR TITLE
Upgrade Spice.ai OSS client and other dependencies

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Upgrade Spice.ai OSS client and other dependencies

### CODE REVIEW

**Review by y-f-u — APPROVED** [review]

File: pkg/plugin/datasource.go [hunk 1]
```diff
@@ -38,25 +38,26 @@ var (
...
 		return nil, fmt.Errorf("http client options: %w", err)
 	}
-	opts.Headers = map[string]string{
-		"Content-Type":    "application/json",
-		"Accept-Encoding": "gzip, deflate",
-		"X-API-Key":       apiKey,
-	}
+	opts.Header.Add("Content-Type", "application/json")
+	opts.Header.Add("Accept-Encoding", "gzip, deflate")
+	opts.Header.Add("X-API-Key", apiKey)
```
Comment (lukekim) [inline pkg/plugin/datasource.go:60]: @sgrebnov let's add a client id string. @phillipleblanc is working on the telemetry spec and can provide the header.

**Review by lukekim — APPROVED** [review]

**Comment by lukekim** [comment]
@sgrebnov do we need to upgrade from DataQuery?

**Comment by sgrebnov** [comment]
> @sgrebnov do we need to upgrade from DataQuery?

Upgraded npm dependencies (including latest grafana) and replaced deprecated DataQuery with version from schema module (as per deprecation/recommendation message).

**Review by lukekim — APPROVED** [review]

File: pkg/plugin/datasource.go (hunk 1 above)
Comment (sgrebnov) [reply pkg/plugin/datasource.go:60]: Update on telemetry  - @phillipleblanc provided me with the future header strings but I've realized that we don't have this logic yet so I can't configure Spice - I'll do this later once Spice supports telemetry / client configuration.

**Pull request merged**